src/use_file.rs

```diff
@@ -68,30 +68,21 @@ fn get_rng_fd() -> Result<libc::c_int, Error> {
 // Succeeds once /dev/urandom is safe to read from
 #[cfg(any(target_os = "android", target_os = "linux"))]
 fn wait_until_rng_ready() -> Result<(), Error> {
-    // Poll /dev/random to make sure it is ok to read from /dev/urandom.
+    // Read a byte from /dev/random to make sure it is ok to read from
+    // /dev/urandom. reading a byte instead of polling is more compatible with
+    // sandboxes that disallow `poll()` but which allow reading /dev/random,
+    // e.g. sandboxes that assume that `poll()` is for network I/O. This way,
+    // fewer applications will have to insert pre-sandbox-initialization logic.
+    // Often (blocking) file I/O is not allowed in such early phases of an
+    // application for performance and/or security reasons.
     let fd = open_readonly(b"/dev/random\0")?;
-    let mut pfd = libc::pollfd {
-        fd,
-        events: libc::POLLIN,
-        revents: 0,
-    };
     let _guard = DropGuard(|| unsafe {
         libc::close(fd);
     });
-
-    loop {
-        // A negative timeout means an infinite timeout.
-        let res = unsafe { libc::poll(&mut pfd, 1, -1) };
-        if res >= 0 {
-            debug_assert_eq!(res, 1); // We only used one fd, and cannot timeout.
-            return Ok(());
-        }
-        let err = crate::util_libc::last_os_error();
-        match err.raw_os_error() {
-            Some(libc::EINTR) | Some(libc::EAGAIN) => continue,
-            _ => return Err(err),
-        }
-    }
+    let mut dummy: [MaybeUninit<u8>; 1] = [MaybeUninit::uninit()];
+    sys_fill_exact(&mut dummy, |buf| unsafe {
+        libc::read(fd, buf.as_mut_ptr().cast::<c_void>(), buf.len())
```

Comment:
I think we can just read it directly without `sys_fill_exact`. `/dev/random` is an "infinite file", so it should never return 0. If it happens anyway, we can return `Error::UNEXPECTED`.

Reply:
> I think we can just read it directly without sys_fill_exact. /dev/random is an "infinite file", so it should never return 0. If it happens anyway, we can return Error::UNEXPECTED.

The processing of the return value of `libc::read` is quite elaborate in `sys_fll_exact`. I'd rather not duplicate any of that. Keep in mind that this is extremely-rarely-executed.

Reply:
I think it can be simplified to:
```rust
let mut buf: [MaybeUninit<u8>; 1] = [MaybeUninit::uninit()];
let ret = libc::read(fd, buf.as_mut_ptr().cast::<c_void>(), buf.len());
match ret {
    1 => Ok(())
    -1 => Err(last_os_error()),
    _ => Err(Error::UNEXPECTED),
}
```

IIRC `EINTR` should not be possible on such small reads.

Reply:
I think keeping the logic the same is probably the most readable. Im fine using `sys_fill_exact` here.

Reply:
I would prefer to keep using `sys_fill_exact`.